doc/delivery-monitoring-protocol.md

```diff
@@ -0,0 +1,51 @@
+# DMP / Delivery Monitoring Protocol
+
+## TERMS
+- `FRAGMENT`: the result of data partitioning.
+- `TRANSPORT`: mechanism of fragmented data interchange that guarantees in-order delivering (examples: [TCP](https://tools.ietf.org/html/rfc793)).
+
+## PROBLEM
+  There are [`TRANSPORT`](#terms)s that can fall and in this case, does not provide information about [`FRAGMENT`](#terms)s were delivered. This creates a problem after returning the ability to transfer fragments: which [`FRAGMENT`](#terms)s should be resended?
+
+## INTRODUCTION
+  [`DMP`](#dmp--delivery-monitoring-protocol) has to be used with the [`TRANSPORT`](#terms) for tracking [`FRAGMENT`](#terms) delivery and, in case of the [`TRANSPORT`](#terms) failure, detecting undelivered [`FRAGMENT`](#terms)s. It solves [`PROBLEM`](#problem), and takes care of the efficient use of memory and the time required for recovery. The protocol is symmetric for the sending and receiving side.
+
+## PROTOCOL DESCRIPTION
+### SPECIAL FRAGMENTS
+- `ACK`: does not contain any information, has to be sent in response to [`ACK`](#special-fragments) from the other side.
+- `SYN`: contains [`PTR`](#data-structures) and [`CNT`](#data-structures), and has to be sent primarily after [`TRANSPORT`](#terms) restoration from both sides.
+  
+### DATA STRUCTURES
+- `S[0]`, `S[1]`: two "storages" of sent [`FRAGMENT`](#terms)s. (The "storage" is only conditional. The main thing is that you should have access to the already sent [`FRAGMENT`](#terms)s, divided into two parts as the protocol requires.)
```

Comment:
What does it mean for "storage" to be conditional? Is "abstract" a better word here?